dev/bots/analyze_snippet_code.dart

```diff
@@ -2,10 +2,61 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// See ../snippets/README.md for documentation.
-
 // To run this, from the root of the Flutter repository:
-//   bin/cache/dart-sdk/bin/dart dev/bots/analyze_snippet_code.dart
+//   bin/cache/dart-sdk/bin/dart --enable-asserts dev/bots/analyze_snippet_code.dart
+
+// In general, please prefer using full inline examples in API docs.
+//
+// For documentation on creating sample code, see ../../examples/api/README.md
+// See also our style guide's discussion on documentation and sample code:
+// https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#documentation-dartdocs-javadocs-etc
+//
+// This tool is used to analyze smaller snippets of code in the API docs.
+// Such snippets are wrapped in ```dart ... ``` blocks, which may themselves
+// be wrapped in {@tool snippet} ... {@endtool} blocks to set them apart
+// in the rendered output.
+//
+// Such snippets:
+//
+//  * If they start with `import` are treated as full application samples; avoid
+//    doing this in general, it's better to use samples as described above. (One
+//    exception might be in dart:ui where the sample code would end up in a
+//    different repository which would be awkward.)
+//
+//  * If they start with a comment that says `// continuing from previous example...`,
+//    they automatically import the previous test's file.
+//
+//  * If they start with a comment that says `// (e.g. in a stateful widget)`,
+//    are analyzed after being inserted into a class that inherits from State.
+//
+//  * If they start with what looks like a getter, function declaration, or
+//    other top-level keyword (`class`, `typedef`, etc), or if they start with
+//    the keyword `final`, they are analyzed directly.
+//
+//  * If they end with a trailing semicolon or have a line starting with a
+//    statement keyword like `while` or `try`, are analyzed after being inserted
+//    into a function body.
+//
+//  * If they start with the word `static`, are placed in a class body before
+//    analysis.
+//
+//  * Otherwise, are used as an initializer for a global variable for the
```

Comment:
Otherwise, _they_ are... ?

Reply:
technically this bullet point continues the sentence started by "Such snippets:", though admittedly by this point in the text the reader has probably forgotten this. :-)